server/bin/pytests/test_logger_type.py

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env python3
+# -*- mode: python -*-
+
+import os, sys
+import logging
+import pytest
+
+from pbench import PbenchConfig, BadConfig, get_pbench_logger
+
+_NAME_ = "pbench-logger-test"
+cfg_name = os.environ["CONFIG"]
+logdir = os.environ["LOGSDIR"]
+
+log_files = {
+    "file": "foo",
+    "devlog": "bar",
+    "hostport": "baz"
+    }
+
+log_msgs = {
+    "file": "logger_type=file in file foo",
+    "devlog": "logger_type=devlog in file bar",
+    "hostport": "logger_type=hostport in file baz"
+}
+
+def mock_the_handler(logger, logger_type, fname):
+    
+    # Assumption: only one Handler is present.
+    hdlr = logger.logger.handlers[0]
+    logger.logger.removeHandler(hdlr)
+
+    # logger.logger is used: the first logger is used to format 
+    # the logs with the help of _styleAdapter and the second is 
+    # used to log the messages
+    fh = logging.FileHandler(os.path.join(logdir,fname))
+    fh.setLevel(logging.DEBUG)
+    logger.logger.addHandler(fh)
+    
+    return logger
+
+def test_pbench_logger():
+
+    config = PbenchConfig(cfg_name)
+    logger = get_pbench_logger(_NAME_, config)
+
+    logger_type = config.get("pbench-server", "logger_type")
+
+    if(logger_type=='file' or logger_type=='devlog' or logger_type=='hostport'):
+        logger.info("Valid logger type")
```

Comment:
We should not do this: it will use the actual logger which might send log messages to some remote service, every time somebody runs the unit test. Just print it.